src/index.js

```diff
@@ -15,7 +15,10 @@ export default function promiseMiddleware({ dispatch }) {
     return isPromise(action.payload)
       ? action.payload.then(
           result => dispatch({ ...action, payload: result }),
-          error => dispatch({ ...action, payload: error, error: true })
+          error => {
+            dispatch({ ...action, payload: error, error: true });
+            return Promise.reject(error);
+          }
```

Comment:
Couldn't that result in a race condition where the user-attached `.then` can run before the `dispatch` call?